superset/databases/api.py

```diff
@@ -909,11 +909,13 @@ def available(self) -> Response:
                 "preferred": engine_spec.engine in preferred_databases,
             }
 
-            if issubclass(engine_spec, BasicParametersMixin):
-                payload["parameters"] = engine_spec.parameters_json_schema()
+            if hasattr(engine_spec, "parameters_json_schema") or hasattr(
```

Comment:
@betodealmeida let me know if this is good enough of a guard statement moving forward to make sure we don't call parameters_json_schema for engines we have implemented yet

Reply:
This looks great!

Reply:
Though I think ideally we'd have all these methods in `BaseEngineSpec` and we'd check if `parameters_schema` is not `None`.